### PR TITLE
Support manual build Polyjuice

### PR DESCRIPTION
**Pull request by Flouse**

### CODE REVIEW

File: docker/layer2/config/scripts-config.json [hunk 1]
```diff
@@ -4,14 +4,13 @@
       "l2_sudt_generator" : "/scripts/godwoken-scripts/sudt-generator",
       "meta_contract_generator" : "/scripts/godwoken-scripts/meta-contract-generator",
       "polyjuice_generator" : "/scripts/godwoken-polyjuice/generator_log",
-
+      "polyjuice_validator" : "/scripts/godwoken-polyjuice/validator",
```
Comment (keroro520) [inline docker/layer2/config/scripts-config.json:7]: Why not use the logging-version validator?

File: docker/.build.mode.env [hunk 2]
```diff
@@ -3,7 +3,7 @@ MANUAL_BUILD_GODWOKEN=false
 MANUAL_BUILD_WEB3=false
 MANUAL_BUILD_WEB3_INDEXER=false
 MANUAL_BUILD_SCRIPTS=false
-MANUAL_BUILD_POLYJUICE=false
+MANUAL_BUILD_POLYJUICE=true
```
Comment (keroro520) [inline docker/.build.mode.env:6]: typo?

File: docker/manual-polyjuice.compose.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  #
+  #   ```
+  #   set -a && source docker/.build.mode.env && ./kicker init
+  #   ```
+  godwoken:
+    volumes:
+      - ./manual-artifacts/polyjuice/validator:/scripts/godwoken-polyjuice/validator
+      - ./manual-artifacts/polyjuice/generator_log:/scripts/godwoken-polyjuice/generator_log
+      - ./manual-artifacts/polyjuice/eth_addr_reg_validator:/scripts/godwoken-polyjuice/eth_addr_reg_validator
+      - ./manual-artifacts/polyjuice/eth_addr_reg_generator:/scripts/godwoken-polyjuice/eth_addr_reg_generator
```
Comment (keroro520) [inline docker/manual-polyjuice.compose.yml:23]: Can we volume the whole directory?

```
volumes:
  - ./manual-artifacts/polyjuice:/scripts/godwoken-polyjuice
```

File: docker/manual-polyjuice.compose.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  # `build` dir to `docker/manual-artifacts/polyjuice`:
+  #
+  #   ```
+  #   set -a && source docker/.build.mode.env && ./kicker init
+  #   ```
+  #
+  # 2. Then start docker-compose service:
+  #
+  #   ```
+  #   set -a && source docker/.build.mode.env && ./kicker init
```
Comment (keroro520) [inline docker/manual-polyjuice.compose.yml:16]:
```suggestion
  #   set -a && source docker/.build.mode.env && ./kicker start
```

**Review by keroro520 — CHANGES_REQUESTED** [review]

File: docker/.build.mode.env (hunk 2 above)
Comment (Flouse) [inline docker/.build.mode.env:6]:
```suggestion
MANUAL_BUILD_POLYJUICE=false
```

File: docker/manual-polyjuice.compose.yml (hunk 3 above)
Comment (Flouse) [reply docker/manual-polyjuice.compose.yml:23]: Yes, it's better.

File: docker/manual-polyjuice.compose.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  #   ```
+  godwoken:
+    volumes:
+      - ./manual-artifacts/polyjuice/validator:/scripts/godwoken-polyjuice/validator
```
Comment (Flouse) [inline docker/manual-polyjuice.compose.yml:20]:
```suggestion
      - ./manual-artifacts/polyjuice:/scripts/godwoken-polyjuice
```

File: docker/manual-polyjuice.compose.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      - ./manual-artifacts/polyjuice/generator_log:/scripts/godwoken-polyjuice/generator_log
```
Comment (Flouse) [inline docker/manual-polyjuice.compose.yml:21]:
```suggestion
```

File: docker/manual-polyjuice.compose.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      - ./manual-artifacts/polyjuice/eth_addr_reg_validator:/scripts/godwoken-polyjuice/eth_addr_reg_validator
```
Comment (Flouse) [inline docker/manual-polyjuice.compose.yml:22]:
```suggestion
```

File: docker/manual-polyjuice.compose.yml (hunk 3 above)
Comment (Flouse) [inline docker/manual-polyjuice.compose.yml:23]:
```suggestion
```

File: docker/layer2/config/scripts-config.json (hunk 1 above)
Comment (Flouse) [reply docker/layer2/config/scripts-config.json:7]: On-chain validator logs will not be output in Godwoken logs.

**Review by keroro520 — APPROVED** [review]

**Pull request merged**